### ISSUE TITLE
Compile crashes with "Permission denied"

### ISSUE THREAD

**Issue by iepetrov** [CERN/TIGRE#23]
Dear Ander Biguri,

Thank you so much for your software!

Unfortunately, I cannot run the "Compile" file. I use Matlab 2014b (newer versions have problems with opengl and nvidia) under Ubuntu 14.04LTS (updated kernel, 64GB RAM), with Nvidia driver 367 and CUDA 8.0. I go to the TIGRE directory, run InitTIGRE and after that I run Compile. Initially it runs ok but it always stops with "Permission denied" - I chmodded all the relevant directories, restarted the computer. I spent more than 4 days trying to find a solution but I could not!

I really need this to work since I am helping with a project of another student and we have to finish by December.

Please HELP!!!

[tigre_problem_19oct2016.txt](https://github.com/CERN/TIGRE/files/540481/tigre_problem_19oct2016.txt)
Thanks in advance!

**Comment by AnderBiguri** [comment]
Hi @iepetrov ,
First of all note: Compile needs to be run only once, if it works then never again. Also, OpenGL is not needed for TIGRE.
It looks like you may not have permission to MATLAB directories, thus compilation fails. You mention that you get "Permision denied" and the .txt you posted has hundreds of errors, all referring to MATLAB mex functions. I suggest you try to compile some basic MATLAB example, to see if the problem resides in TIGRE or in your local configuration.  [You can find some examples in the documentation](https://uk.mathworks.com/help/matlab/matlab_external/table-of-mex-file-source-code-files.html).

If you can not compile these, then you need to make sure you have access to the libraries on you computer, and I can not help with that.  If you can, then TIGRE may have a problem and we need to look into it further. 

Let me know
Ander

**Comment by iepetrov** [comment]
Thank you so much for the fast reply!!! I will try that tomorrow and I will let you know. Thanks

**Comment by iepetrov** [comment]
Dear Ander,

I run one of the examples from the Matlab page you recommended. It was a success - "MEX completed successfully."
[mex_test_20oct2016.txt](https://github.com/CERN/TIGRE/files/542333/mex_test_20oct2016.txt)

I am trying to promote your software in our group and eventually the other groups (quite a lot imaging people here), so it is important for it to run smoothly. And that is additional to the project I will try to use it for.

Thanks in advance.

**Comment by AnderBiguri** [comment]
@iepetrov, I see, so you do have access to run mex files. Then the problem must be somewhere else.

Are you sure that you .xml file is correctly configured? 

https://github.com/CERN/TIGRE/issues/15

And 

https://github.com/CERN/TIGRE/issues/4

Have a bit more insight about possible solutions. Also, how do you get the  "Permission denied" error? what throws that? Is MATLAB erroring there? 

Also, for some reason your txt seems to be opening parallel pools to Compile, is that right? I am not sure if that is a good idea. As another possible error: You log says that you are using gcc 4,8,4 while the supported one is 4.7.x. I am quite surprised that you managed to compile with an unsupported compiler, as MATLAB is very very picky about compiler versions. Try using that specific compiler version.

I am very happy to hear that you are trying to support TIGRE!  It depends how far your group is from Bath, I could visit and give a talk/tutorial/demo on TIGRE, if you are interested.

Please try the suggestions I proposed and come back to me. I'd say the most important suggestion is use a supported compiler.
Hope you make it work
Ander

**Comment by iepetrov** [comment]
Dear Ander Biguri,

I think I changed the two lines in in the XML file but maybe I haven't done everything correctly
[xml_20oct2016.txt](https://github.com/CERN/TIGRE/files/542450/xml_20oct2016.txt)
. I will attach the file. I am Canada.

Thanks

**Comment by AnderBiguri** [comment]
@iepetrov unfortunately I can not tell if the xml is wrong without the computer itself. 
Please try the rest of the suggestions.
Lets see if we can make this work!
Best,
Ander

**Comment by AnderBiguri** [comment]
@iepetrov Did you fix your problems?

**Comment by iepetrov** [comment]
Dear Ander Biguri,

I was able to fix it. In l. 44 in the xml file I had to remove a semicolon and to add '-L' (I had entered the correct paths before that but that was not the problem).

The demo run ok - DEMO 9:  Algorithms 04.

When I decided to run with my data I encountered problems. I had to transpose the angle file in 'order_subset.m'. I had to permute the proj in 'OS_SART.m' and 'ASD_POSC.m' - in the example I run your projections are nxn while in my case they are mxn so I had to permute them. Unfortunately, the whole MATLAB crashes when it reaches  minTV.cpp. It crashes either with segmentation error or, more frequently, malloc error ( *** Error in `/usr/local/MATLAB/R2014b/bin/glnxa64/MATLAB': malloc(): memory corruption: 0x00007fb789680040 ***) - I read something about not allocating enough memory ... but I am not a cpp guy so I really do not know how to fix this.

Thanks for your help and for following up.

**Comment by AnderBiguri** [comment]
@iepetrov  Im glad you fixed it! Im going to close this issue on github, but please, lets continue this discussion via email.

I will definitely have a look at that non-square detectors. Most of the tests I performed had square detectors, so I might have missed something.  About the minTV, could you please tell me what size of image, detector, number of projections, and what GPU you have?  minTV is memory intensive, you may be running out of memory. It would be good to know if that's the case.

Best
Ander

**Comment by AnderBiguri** [comment]
Hi again @iepetrov ,

I tested Os_SART and ASD_POCS with different sizes of detector and they seem to work fine. Is it possible that you had your data transposed outside?If you still think its wrong, could you please help me giving me a minimal code that would reproduce the error?

Additionally, there are some errors in minTV regarding image size (when image is not a perfect cube), but otherwise, you'd need to check the size of the image you are using and the memory availabl ein your GPU, as you may just get that error due to not having enough RAM
Ander
